cmd/guacrest/cmd/server.go

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright 2024 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/go-chi/chi"
+	gen "github.com/guacsec/guac/pkg/guacrest/generated"
+	"github.com/guacsec/guac/pkg/guacrest/server"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+// TODO: add logging middleware
+// TODO: add context propagation middleware
+
+func startServer() {
+	ctx := logging.WithLogger(context.Background())
+	logger := logging.FromContext(ctx)
+
+	gqlClient := getGraphqlServerClientOrExit(ctx)
+	handler := server.NewDefaultServer(gqlClient)
+	handlerWrapper := gen.NewStrictHandler(handler, nil)
+	router := chi.NewRouter()
+	router.Mount("/", gen.Handler(handlerWrapper))
+	server := http.Server{
+		Addr:    fmt.Sprintf(":%d", flags.restAPIServerPort),
+		Handler: router,
+	}
+
+	proto := "http"
+	if flags.tlsCertFile != "" && flags.tlsKeyFile != "" {
+		proto = "https"
+	}
+
+	logger.Infof("Connect to the server at %s://localhost:%d/", proto, flags.restAPIServerPort)
```

Comment:
should be `0.0.0.0` maybe instead of localhost? localhost generally is routed to 127.0.0.1 which is only accessible locally

Reply:
Done